spec/requests/products_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :request do
+
+  describe 'POST products#add_to_cart' do
+    let(:product) { create :product }
+    subject { post add_to_cart_product_path(product.id) }
+
+    context 'not logged in' do
```

Comment:
I think you could also add a test that succeeds, not only for unauthorized users